enumerable.rb

```diff
@@ -0,0 +1,99 @@
+module Enumerable
+  def my_each(array)
+    array.length.times { |index| yield(array[index]) }
+
+    nil
+  end
+
+  def my_each_with_index(array)
+    0.upto(array.length - 1) { |index| yield(array[index], index) }
+
+    nil
+  end
+
+  def my_select(array)
+    result = []
+
+    my_each(array) { |item| result << item if yield(item) }
+
+    result
+  end
+
+  def my_all?(array)
+    result = true
+
+    my_each(array) { |item| result = false unless yield(item) }
+
+    result
+  end
+
+  def my_any?(array)
+    result = false
+
+    my_each(array) { |item| result = true if yield(item) }
+
+    result
+  end
+
+  def my_none?(array)
+    result = true
+
+    my_each(array) { |item| result = false if yield(item) }
+
+    result
+  end
+
+  def my_count(array, value = nil)
+    counter = 0
+
+    my_each(array) do |item|
+      if block_given?
+        counter += 1 if yield(item)
+      elsif item == value
+        counter += 1
+      else
+        counter = array.length unless value
+      end
+    end
+    counter
+  end
```

Comment:
* [ ] should have working `my_count` identical and returns the same thing as ruby's [`count`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-count). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_count.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_count

- [ ] my_count returns the number of items in enum through enumeration
```ruby
[5, 4, 1].my_count == [5, 4, 1].count => 3
```
- Yours is giving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/77696262-b4e3bc00-6f72-11ea-9cbe-75dc178d2de8.png)

- Real count
![image](https://user-images.githubusercontent.com/52209067/77696292-c200ab00-6f72-11ea-8e97-5a2b79256cbc.png)

- [ ] my_count counts the number of items in enum that are equal to the item if an argument is given
```ruby
[5, 4, 1].my_count(4) == [5, 4, 1].count(4) => 1
```
- Yours is giving an NoMethodError
![image](https://user-images.githubusercontent.com/52209067/77696347-d6dd3e80-6f72-11ea-8b7e-60078c7c0085.png)

- Real count
![image](https://user-images.githubusercontent.com/52209067/77696378-e2306a00-6f72-11ea-9310-0bb30ab9c94d.png)

- [ ] my_count counts the number of elements yielding a true value if a block is given
```ruby
[5, 4, 1].my_count { |x| x > 2 } == [5, 4, 1].count { |x| x > 2 }
```
- Yours is giving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/77696427-faa08480-6f72-11ea-8113-b3fbc2fd1a81.png)

- Real count
![image](https://user-images.githubusercontent.com/52209067/77696440-055b1980-6f73-11ea-9b41-09dac295d3d3.png)